presto-spi/src/main/java/io/prestosql/spi/security/ViewExpression.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.spi.security;
+
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public class ViewExpression
+{
+    private final String identity;
+    private final Optional<String> catalog;
+    private final Optional<String> schema;
+    private final String expression;
+
+    public ViewExpression(String identity, Optional<String> catalog, Optional<String> schema, String expression)
+    {
+        this.identity = requireNonNull(identity, "identity is null");
+        this.catalog = requireNonNull(catalog, "catalog is null");
+        this.schema = requireNonNull(schema, "schema is null");
+        this.expression = requireNonNull(expression, "expression is null");
+
+        if (!catalog.isPresent() && schema.isPresent()) {
+            throw new IllegalArgumentException("catalog must be present if schema is present");
+        }
+    }
+
+    public String getIdentity()
+    {
+        return identity;
+    }
+
+    public Optional<String> getCatalog()
```

Comment:
I don't think that connector should return `catalog`. Connector is only aware of catalog name it is using, don't know anything about other catalogs. Do we need `ConnectorViewExpression` and `SystemViewExpression`?

Reply:
The catalog/schema/identity make up the "environment" under which the filter or mask should be evaluated. They are needed for the purpose of resolving relative table references, and it's exactly the same mechanism we use for views. In a way, row filters and masks behave like dynamic views.